### PR TITLE
Gcb allow build step failure

### PR DESCRIPTION
**Pull request by arvinddayal**
Add allowFailure and allowExitCodes to CloudBuild Trigger.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudbuild: Added `allow_failure` and `allow_exit_codes` to `build.step` in `google_cloudbuild_trigger` resource
```

### CODE REVIEW

File: mmv1/products/cloudbuild/Trigger.yaml [hunk 1]
```diff
@@ -978,6 +978,23 @@ properties:
...
                 When script is provided, the user cannot specify the entrypoint or args.
+            - !ruby/object:Api::Type::Boolean
+              name: 'allowFailure'
+              description: |
+                Allow this build step to fail without failing the entire build.
+                If false, the entire build will fail if this step fails. Otherwise, the
+                build will succeed, but this step will still have a failure status.
+                Error information will be reported in the `failureDetail` field.
+
+                `allowExitCodes` takes precedence over this field.
```
[the remark this answers is not in the document]
Comment (arvinddayal) [reply mmv1/products/cloudbuild/Trigger.yaml:989]: Updated

File: mmv1/templates/terraform/examples/cloudbuild_trigger_allow_failure.tf.erb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  trigger_template {
+    branch_name = "main"
+    repo_name   = "my-repo"
+  }
+
+  build {
+    step {
+      name = "ubuntu"
+      args = ["-c", "exit 1"]
+      allowFailure = true # using allowFailure field
```
Comment (zli82016) [inline mmv1/templates/terraform/examples/cloudbuild_trigger_allow_failure.tf.erb:13]: It should be `allow_failure = true` and the comment for this field is not necessary.

File: mmv1/templates/terraform/examples/cloudbuild_trigger_allow_exit_codes.tf.erb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  trigger_template {
+    branch_name = "main"
+    repo_name   = "my-repo"
+  }
+
+  build {
+    step {
+      name = "ubuntu"
+      args = ["-c", "exit 1"]
+      allowExitCodes = [1] # using allowExitCodes field
```
Comment (zli82016) [inline mmv1/templates/terraform/examples/cloudbuild_trigger_allow_exit_codes.tf.erb:13]: It should be `allow_exit_codes` and the comment is not necessary

File: mmv1/templates/terraform/examples/cloudbuild_trigger_allow_exit_codes.tf.erb [hunk 4]
```diff
@@ -0,0 +1,66 @@
+resource "google_cloudbuild_trigger" "<%= ctx[:primary_resource_id] %>" {
```
Comment (zli82016) [inline mmv1/templates/terraform/examples/cloudbuild_trigger_allow_exit_codes.tf.erb:1]: This  example needs to be added to the file mmv1/products/cloudbuild/Trigger.yaml.

This is an example https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/cloudbuild/Trigger.yaml#L39

File: mmv1/templates/terraform/examples/cloudbuild_trigger_allow_exit_codes.tf.erb (hunk 3 above)
Comment (zli82016) [reply mmv1/templates/terraform/examples/cloudbuild_trigger_allow_exit_codes.tf.erb:13]: Also, can you add more items to allow_exit_codes, as it is an array?

**Comment by zli82016** [comment]
Is there an Github issue for this feature request?

**Comment by arvinddayal** [comment]
There's no GH feature request for this. I'm adding this now that the following features have been made generally available: https://cloud.google.com/build/docs/build-config-file-schema#allowfailure and https://cloud.google.com/build/docs/build-config-file-schema#allowexitcodes

**Review by zli82016 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**